lib/health.js

```diff
@@ -0,0 +1,42 @@
+'use strict';
+
+const request = require('request-promise-native');
+
+const hasMatchingMethod = (option, reqMethod) => {
+  return option.methods.some(method => {
+    return method.toLowerCase() === reqMethod.toLowerCase();
+  });
+};
+
+module.exports = endpoints => {
+
+  return (req, res, next) => {
+
+    if (endpoints && endpoints.length) {
+
+      let requests = endpoints.reduce((memo, option) => {
+        // Test the method property against the current req method
+        if (option.methods && (hasMatchingMethod(option, req.method) !== true)) {
+          return memo;
+        }
+        // memoize the endpoints
+        memo.push(request(option.url).promise());
+        return memo;
+      }, []);
```

Comment:
I'd go for only GET. Either a POST doesn't talk to services, and will immediately 302 to a GET anyway, or it will talk to services, and we just end up hitting the service to check if hitting the service will work, which seems a bit redundant, when we can just "try to do the thing".

Either way, checking on GET and POST looks like it will just result in making twice as many requests as we need.